src/System.Configuration.ConfigurationManager/tests/System/Configuration/StringValidatorTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Configuration;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public class StringValidatorTests
+    {
+        [Fact]
+        public void Constructor_MinLength()
+        {
+            //We don't expect this to fail
+            StringValidator validator = new StringValidator(5);
+        }
+
+        [Fact]
+        public void Constructor_MinValueAndMaxValue()
+        {
+            // This should complete with no errors.
+            StringValidator validator = new StringValidator(5, 10);
+        }
+
+        [Fact]
+        public void Constructor_MinValueMaxValueAndInvalidChars()
+        {
+            // this should complete with no errors.
+            StringValidator validator = new StringValidator(5, 10, "abcde");
+        }
+
+        [Fact]
+        public void CanValidate_PassInStringType()
+        {
+            StringValidator validator = new StringValidator(5);
+            bool result = validator.CanValidate(typeof(string));
+            Assert.True(result);
+        }
+
+        [Fact]
+        public void CanValidate_PassInNotStringType()
+        {
+            StringValidator validator = new StringValidator(5);
+            bool result = validator.CanValidate(typeof(Int32));
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void Validate_PassInNonString()
+        {
+            StringValidator validator = new StringValidator(5);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate(5));
+            ArgumentException expected = new ArgumentException(SR.Validator_value_type_invalid, string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
+        }
+
+        [Fact]
+        public void Validate_StringTooSmall()
+        {
+            StringValidator validator = new StringValidator(5);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate("Hi"));
+            ArgumentException expected = new ArgumentException(string.Format(SR.Validator_string_min_length, 5), string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
+        }
+
+        [Fact]
+        public void Validate_StringTooBig()
+        {
+            StringValidator validator = new StringValidator(5, 10);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate("This is more than ten"));
+            ArgumentException expected = new ArgumentException(string.Format(SR.Validator_string_max_length, 10), string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
```

Comment:
Well you are doing it the right way. 
```cs
Assert.Throws<ArgumentException>(() => validator.Validate("This is more than ten"));
```
Will actually cover what you want to test as if there is no exception thrown or the exception is different than an `ArgumentException` this test will fail. 

What I meant saying we don't assert for exception messages is actually asserting for exception.Message since the message can change in between platforms and can be localized in the future. So we shouldn't do:
```cs
ArgumentException expected = new ArgumentException(string.Format(SR.Validator_string_max_length, 10), string.Empty);
Assert.Equal(expected.Message, thrownException.Message);
```